### PR TITLE
Add support to specify maxDelay parameter

### PR DESCRIPTION
**Pull request by code-haven**
So at work, we were experimenting with this plugin to connect Grafana to SignalFX. This worked really well (Thanks!) except that we found that some graphs(with multiple queries) were ~5-10 minutes behind current time. We compared this to our existing dashboards and figured out that `maxDelay` was set explicitly and couldn't on Grafana.

This PR adds support to specify `maxDelay` *per query*.

However, I'm not sure if this is the best way to do it. Because AFAIK, this plugin merges all queries in the chart before executing it - which makes this setting is more of a *chart* level parameter rather than a *per query* parameter. Is there a way to add a parameter at a chart level rather than at a query level? Or is the current approach good enough?

NOTE:
1. I had to bump the version of signalfx-go to get the backend code to build.

### CODE REVIEW

File: src/stream_handler.js [hunk 1]
```diff
@@ -27,8 +27,9 @@ export class StreamHandler {
     this.templateSrv = templateSrv;
   }
 
-  start(program, aliases, options) {
+  start(program, aliases, maxDelay, options) {
     this.aliases = aliases;
+    this.maxDelay = maxDelay;
```
Comment (piotr-signalfx) [inline src/stream_handler.js]: this.maxDelay should be set to 0 in case the provided maxDelay is undefined.

File: src/datasource.js [hunk 2]
```diff
@@ -54,6 +55,12 @@ export class SignalFxDatasource {
       .flatMap(t => this.extractLabelsWithAlias(t.program, t.alias)));
   }
 
+  getMaxDelay(options) {
+    const target = _.max(options.targets, t => { return t.maxDelay });
```
Comment (piotr-signalfx) [inline src/datasource.js]: Probably can be shortened to something like: `_.max(_.map(options.targets, t => t.maxDelay));`

File: src/stream_handler.js (hunk 1 above)
Comment (code-haven) [reply src/stream_handler.js]: If `maxDelay` is not passed to SignalFX, wouldn't it default to some value? Will passing it as `0` have the same effect?

Comment (piotr-signalfx) [reply src/stream_handler.js]: this.maxDelay is also used to calculate nextEstimatedTimestamp, which might happen before the actual value is received in the JOB_INITIAL_MAX_DELAY message.

Comment (code-haven) [reply src/stream_handler.js]: fair enough. I have pushed the change for this.

**Comment by piotr-signalfx** [comment]
The functionality has been introduced via https://github.com/signalfx/grafana-signalfx-datasource/pull/19

**Comment by piotr-signalfx** [comment]
@code-haven Thanks for your contribution

**Pull request closed without merging**